src/fabric_cli/commands/fs/export/fab_fs_export_item.py

```diff
@@ -99,8 +101,18 @@ def export_single_item(
 
         args.from_path = item.path.strip("/")
         args.ws_id, args.id, args.item_type = workspace_id, item_id, str(item_type)
-        args.format = definition_format_mapping.get(item_type, "")
-
+        if _export_format := getattr(args, "format", None):
+            if _export_format not in (".py", ".ipynb"):
+                raise FabricCLIError(
+                    "Invalid format. Only '.py' and '.ipynb' are supported.",
+                    fab_constant.ERROR_INVALID_INPUT,
+                )
+            elif _export_format == ".py":
+                args.format = "?format=fabricGitSource"
+            else:
+                args.format = "?format=ipynb"
+        else:
+            args.format = definition_format_mapping.get(item_type, "")
         item_def = item_api.get_item_withdefinition(args, item_uri)
 
```

Comment:
Please consider enhancing definition_format_mapping to return an object that includes the supported format type as below:
```suggestion
definition_format_mapping = {
    ItemType.SPARK_JOB_DEFINITION: {
        "default": "?format=SparkJobDefinitionV1"
    },
    ItemType.NOTEBOOK: {
        "default": "?format=ipynb",
        ".py": "?format=fabricGitSource", 
        ".ipynb": "?format=ipynb"
    },
}

```
With that value, we can eliminate the current if/else branching.
Additionally, we can set the format param input’s default to "default" so we always have a defined value.